components/mediation/security/org.wso2.micro.integrator.mediation.security/src/main/java/org/wso2/micro/integrator/mediation/security/vault/SecretCipherHander.java

```diff
@@ -29,15 +29,10 @@
 	private static Log log = LogFactory.getLog(SecretCipherHander.class);
 
 	/* Root Secret Repository */
-	private RegistrySecretRepository parentRepository = new RegistrySecretRepository();
+	private CiphertextRepository parentRepository = CiphertextRepository.getInstance();
```

Comment:
Is it possible to initialize an object and use it here? Or is there a specifc reason to use a singleton?

Reply:
Ideally we should not be creating secret repository objects for each and every vault:lookup. This is to limit repository object creation.